esphome/components/esp32/gpio.h

```diff
@@ -21,17 +27,26 @@ class ESP32InternalGPIOPin : public InternalGPIOPin {
   std::string dump_summary() const override;
   void detach_interrupt() const override;
   ISRInternalGPIOPin to_isr() const override;
-  uint8_t get_pin() const override { return (uint8_t) pin_; }
+  uint8_t get_pin() const override { return pin_; }
   gpio::Flags get_flags() const override { return flags_; }
-  bool is_inverted() const override { return inverted_; }
+  bool is_inverted() const override { return pin_flags_.inverted; }
+  gpio_num_t get_pin_num() const { return static_cast<gpio_num_t>(pin_); }
+  gpio_drive_cap_t get_drive_strength() const { return static_cast<gpio_drive_cap_t>(pin_flags_.drive_strength); }
 
  protected:
   void attach_interrupt(void (*func)(void *), void *arg, gpio::InterruptType type) const override;
 
-  gpio_num_t pin_;
-  gpio_drive_cap_t drive_strength_;
-  gpio::Flags flags_;
-  bool inverted_;
+  // Memory layout: 8 bytes total on 32-bit systems
+  // - 3 bytes for members below
+  // - 1 byte padding for alignment
+  // - 4 bytes for vtable pointer
+  uint8_t pin_;        // GPIO pin number (0-255, actual max ~48 on ESP32)
```

Comment:
The ESP32-P4 chip features 55 physical GPIO pins (GPIO0 ~ GPIO54).
```suggestion
  uint8_t pin_;        // GPIO pin number (0-255, actual max ~54 on ESP32)
```